### PR TITLE
Fix TSX compiling when running JEST tests

### LINKED ISSUE #259: Devs SBAT test the website
Expected Behavior
We run tests

Current Behavior
We don't run tests

### PR DESCRIPTION
**Pull request by aaronmgdr**
### Description

Updates our ts jest config and related configs so that TSX files were processed. 

Also add react testing library

### Tested

wrote a new two snapshot tests and a few tests using react testing lib

### Other changes

fixes typing on Button.tsx
add type param to withScreenSize
other typing improvements

### Related issues

- Fixes #259 

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: packages/web/jest.config.js [hunk 1]
```diff
@@ -4,8 +4,23 @@ module.exports = {
...
+    'next-i18next': '<rootDir>/node_modules/next-i18next',
+    'next/config': '<rootDir>/__mocks__/next-server/config',
+  },
+  modulePathIgnorePatterns: ['<rootDir>/node_modules/(.*)/node_modules/react-native'],
   preset: './node_modules/react-native-web/jest-preset.js',
-  testEnvironment: 'node',
+  setupFiles: ['jest-canvas-mock'],
+  setupFilesAfterEnv: ['<rootDir>/jest_setup.ts'],
+  testEnvironment: 'jsdom',
+  transformIgnorePatterns: ['node_modules/(?!react-native|react-navigation|)'],
```
Comment (cmcewen) [inline packages/web/jest.config.js]: Take a look at jean's recent changes to the other jest.config files for this one

File: packages/web/server/airtable.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ export default function airtableInit(baseID: string) {
   const { serverRuntimeConfig } = getConfig()
 
   if (!airTableSingleton) {
-    airTableSingleton = new AirtableAPI({ apiKey: serverRuntimeConfig.AIRTABLE_API_KEY })
+    // tslint:disable-next-line
+    airTableSingleton = new Airtable({ apiKey: serverRuntimeConfig.AIRTABLE_API_KEY })
```
Comment (cmcewen) [inline packages/web/server/airtable.ts:10]: Are the types wrong here?

File: packages/web/server/index.ts [hunk 3]
```diff
@@ -20,6 +21,7 @@ import { getFormattedMediumArticles } from './mediumAPI'
 const port = parseInt(process.env.PORT, 10) || 3000
 
 const dev = process.env.NEXT_DEV === 'true'
+// tslint:disable-next-line
```
Comment (cmcewen) [inline packages/web/server/index.ts]: What is wrong with this?

File: packages/web/tsconfig.json [hunk 4]
```diff
@@ -2,7 +2,8 @@
   "compileOnSave": false,
```
Comment (cmcewen) [inline packages/web/tsconfig.json:2]: can you extend our core shared file here or no?

File: packages/web/server/airtable.ts (hunk 2 above)
Comment (aaronmgdr) [reply packages/web/server/airtable.ts:10]: ah these must have been left over from a trip down a different rabbit hole

File: packages/web/tsconfig.json (hunk 4 above)
Comment (aaronmgdr) [reply packages/web/tsconfig.json:2]: hmm not without having to change code to fix a lot of new compile errors

File: packages/web/server/index.ts (hunk 3 above)
Comment (aaronmgdr) [reply packages/web/server/index.ts]: ah these must have been left over from a trip down a different rabbit hole

**Pull request closed without merging**